iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/IamOAuth2RequestFactory.java

```diff
@@ -164,4 +186,17 @@ public TokenRequest createTokenRequest(Map<String, String> requestParameters,
     return new TokenRequest(requestParameters, clientId, scopes, grantType);
   }
 
+  private boolean isValidUrl(String url) {
```

Comment:
There is no need to return `boolean` (or anything else), since in case of failure an exception is raised (but I don't know how `forEach` behaves in case of a function not returning anything.